core/server/api_container/server/startosis_engine/startosis_packages/git_package_content_provider/git_package_content_provider_test.go

```diff
@@ -188,19 +189,55 @@ func TestGetAbsolutePathOnDisk_WorksForNonInMainBranchLocators(t *testing.T) {
 	packageDir, err := os.MkdirTemp("", packagesDirRelPath)
 	require.Nil(t, err)
 	defer os.RemoveAll(packageDir)
-	packageTmpDir, err := os.MkdirTemp("", packagesTmpDirRelPath)
+	packageTmpDir, err := os.MkdirTemp("", repositoriesTmpDirRelPath)
 	require.Nil(t, err)
 	defer os.RemoveAll(packageTmpDir)
 
 	provider := NewGitPackageContentProvider(packageDir, packageTmpDir, nil)
 
 	absoluteFileLocator := "github.com/kurtosis-tech/sample-dependency-package@test-branch/main.star"
-	pathOnDisk, err := provider.GetOnDiskAbsoluteFilePath(absoluteFileLocator)
+	pathOnDisk, err := provider.getOnDiskAbsolutePath(absoluteFileLocator, true)
 
 	require.Nil(t, err, "This test depends on your internet working and the kurtosis-tech/datastore-army-package existing")
 	require.Equal(t, path.Join(packageDir, "kurtosis-tech", "sample-dependency-package", "main.star"), pathOnDisk)
 }
 
+func TestGetAbsolutePathOnDisk_GenericRepositoryDir(t *testing.T) {
```

Comment:
Should we also add an e2e test for upload_files?

Reply:
it was added, thanks.